Wrappers/Python/cil/io/RAWFileWriter.py

```diff
@@ -0,0 +1,145 @@
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.import numpy as np
+
+from cil.framework import AcquisitionData, ImageData
+import os, re
+import sys
+from cil.framework import AcquisitionData, ImageData
+import re
+from cil.io import utilities
+import configparser
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+def compress_and_save(data, compress, scale, offset, dtype, fname):
+    '''Compress and save numpy array to file
+    
+    Parameters
+    ----------
+    data : numpy array
+    scale : float
+    offset : float
+    dtype : numpy dtype
+    fname : string
+    '''
+    if compress:
+        d = utilities.compress_data(data, scale, offset, dtype)
+    else:
+        d = data
+    # Data is always written in ‘C’ order, independent of the order of d. 
+    d.tofile(fname)    
+
+class RAWFileWriter(object):
+    '''
+        Writer to write DataSet to disk as a binary blob
+
+        This writer will write a text file with the minimal information necessary to 
+        read the data back in.
+        
+        Parameters
+        ----------
+        data : DataContainer, AcquisitionData or ImageData
+            This represents the data to save to TIFF file(s)
+        file_name : string
+            This defines the file name prefix, i.e. the file name without the extension.
+        compression : str, default None. Accepted values None, 'uint8', 'uint16'
+            The lossy compression to apply. The default None will not compress data. 
+            'uint8' or 'unit16' will compress to unsigned int 8 and 16 bit respectively.
+
+
+        Note:
+        -----
+
+        If compression 'uint8' or 'unit16' are used, the scale and offset used to compress the data are saved 
+        in a file called `scaleoffset.json` in the same directory as the TIFF file(s).
+
+        The original data can be obtained by: `original_data = (compressed_data - offset) / scale`
+    '''
+    
+    def __init__(self,
+                 **kwargs):
+        
+        
+        self.data_container = kwargs.get('data', None)
+        self.file_name = kwargs.get('file_name', None)
+        compression = kwargs.get('compression', None)
+        
+        if ((self.data_container is not None) and (self.file_name is not None)):
+            self.set_up(data = self.data_container,
+                        file_name = self.file_name, 
+                        compression=compression)
+        
+    def set_up(self,
+               data = None,
+               file_name = None,
+               compression=0):
+        
+        self.data_container = data
+        file_name = os.path.abspath(file_name)
+        self.file_name = os.path.splitext(
+            os.path.basename(
+                file_name
+                )
+            )[0]
```

Comment:
Looking at this upset my eyes
```suggestion
        self.file_name = os.path.splitext(
            os.path.basename(file_name)
            )[0]
```